hyperspy/signal.py

```diff
@@ -4920,6 +4932,84 @@ def T(self):
+            If None, first order hann window is used.
+            Higher orders result in more homogeneous intensity distribution.
+        tukey_alpha : float
+            Only used if window='tykey'
+            From scipy documentation:
+            Shape parameter of the Tukey window, representing the fraction of
+            the window inside the cosine tapered region. If zero,
+            the Tukey window is equivalent to a rectangular window.
+            If one, the Tukey window is equivalent to a Hann window.
+            (Default: 0.5)
+        inplace : boolean, optional
+            If True apodization applied in place, i.e. signal data will be substituted by the apodized one.
+            (Default: False)
+        Returns
+        -------
+        out : :class:`~hyperspy.signals.BaseSignal or subclasses, optional
+            If 'inplace'=True, returns apodized signal of the same type as self.
+        Examples
+        --------
+        >>> import hyperspy.api as hs
+        >>> holo = hs.datasets.example_signals.object_hologram()
+        >>> holo.apply_apodization('tukey', tukey_alpha=0.1).plot()
+        """
+
+        if window == 'hanning' or window == 'hann':
+            if hann_order:
+                window_function = lambda m: hann_window_nth_order(m, hann_order)
+            else:
+                window_function = lambda m: np.hanning(m)
+        elif window == 'hamming':
+            window_function = lambda m: np.hamming(m)
+        elif window == 'tukey':
+            window_function = lambda m: sp.signal.tukey(m, tukey_alpha)
+        else:
+            raise ValueError('Wrong type parameter value.')
+
+        windows_1d = []
+
+        axes = np.array(self.axes_manager.signal_indices_in_array)
+
+        for axis, axis_index in zip(self.axes_manager.signal_axes, axes):
+            if isinstance(self.data, da.Array):
+                chunks = self.data.chunks[axis_index]
+                window_da = da.from_array(window_function(axis.size),
+                                          chunks=(chunks, ))
+                windows_1d.append(window_da)
+            else:
+                windows_1d.append(window_function(axis.size))
+
+        window_nd = outer_nd(*windows_1d).T
+
+        # Prepare slicing for multiplication window_nd nparray with data with higher dimensionality:
+        if inplace:
+            slice_w = []
+
+            # Iterate over all dimensions of the data
+            for i in range(self.data.ndim):
+                if any(i == axes):  # If current dimension represents one of signal axis, all elements in window
+                    # along current axis to be subscribed
+                    slice_w.append(slice(None))
+                else:  # If current dimension is navigation one, new axis is absent in window and should be created
+                    slice_w.append(None)
+
+            self.data = self.data * window_nd[tuple(slice_w)]
```

Comment:
```suggestion
            self.data = self.data * window_nd[tuple(slice_w)]
            self.events.data_changed.trigger(obj=self)
```
Trigger an event when data is changed in place.